src/scopes.js

```diff
@@ -56,3 +61,47 @@ exports.scopeMatch = function(scopePatterns, scopesets) {
     });
   });
 };
+
+/**
+ * Finds scope intersections between scopePatterns and scope-set
+ *
+ * Note that scope-set is an array of arrays of strings on negation-free
+ * disjunctive normal form.
+ *
+ */
+exports.scopeIntersection = (scopePatterns, scopesets) => {
```

Comment:
I think we just want to intersect two scopesets:
```javascript
exports.scopeIntersection = (scopeset1, scopeset2) => { /* ... */ };

exports.scopeIntersection(
  ["queue:*"],
  ["auth:create-client:foo.bar', 'queue:create-task:foo/bar"]
);
```

The existing comments in this file are incorrect (or at least misleading): a scopeset is an array of scopes, so the comment should say "Note that scopesets is an array of arrays of strings" (which is to say, `scopesets` is an array of `scopeset`s).